src/app/containers/AssignmentTemplateEditForm.tsx

```diff
@@ -29,7 +29,8 @@ export interface AssignmentTemplateEditFormProps extends AssignmentFormProps{
 
 const mapStateToProps = (state: RootState, props: AssignmentTemplateEditFormProps) => {
     if (state && state.assignments && state.assignments.templates) {
-        const initialTemplate = state.assignments.templates[props.id];
+        const initialTemplateIndex = state.assignments.templates.findIndex((value) => value.id==props.id);
```

Comment:
I would create a Select for this in the selectors file:

```ts
export const getAssignmentTemplate = (id?:number)=> (state:RootState)=>{
    if(id==null){
        return null;
    }
    return allTemplates(state).find(t=>t.id==id);
}
```
Then later you'd import your selector into this file and use it:

```ts
const mapStateToProps = (state:RootState, props:AssignmentTemplateEditFormProps)=>{
  return {
    initialValues: getAssignmentTemplate(id)(state)
 }
}
```

This moves all of the selection logic into it's own maintained file which makes it easier to maintain.  Also if we end up changing the way the templates are stored, from a list structure to a map we only have to change the selector not everywhere that is using that piece of state.